libraries/hermes/src/Proxy/Proxy.ts

```diff
@@ -0,0 +1,89 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import { ConduitRouter } from '../Router';
+import ConduitGrpcSdk from '@conduitplatform/grpc-sdk';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+import { ProxyRoute, TypeRegistry } from '../classes';
+
+export class ProxyRouteController extends ConduitRouter {
+  private _proxyRoutes: Map<string, ProxyRoute>;
+  private globalMiddlewares: ((
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) => void)[];
+
+  constructor(
+    grpcSdk: ConduitGrpcSdk,
+    private readonly metrics?: {
+      registeredRoutes?: {
+        name: string;
+      };
+    },
+  ) {
+    super(grpcSdk);
+    this._proxyRoutes = new Map();
+    this.globalMiddlewares = [];
+    this.initializeRouter();
+  }
+
+  private initializeRouter() {
+    this.createRouter();
+    this._expressRouter!.use((req: Request, res: Response, next: NextFunction) => {
+      this.globalMiddlewares.forEach(middleware => middleware(req, res, next));
+    });
+  }
+
+  registerGlobalMiddleware(
+    middleware: (req: Request, res: Response, next: NextFunction) => void,
+  ) {
+    this.globalMiddlewares.push(middleware);
+  }
+
+  registerProxyRoute(route: ProxyRoute) {
+    const key = `${route.input.path}-${route.input.target}`;
+    const registered = this._proxyRoutes.has(key);
+    this._proxyRoutes.set(key, route);
+    if (registered) {
+      this.refreshRouter();
+    } else {
+      this.addProxyRoute(route);
+      if (this.metrics?.registeredRoutes) {
+        ConduitGrpcSdk.Metrics?.increment(this.metrics.registeredRoutes.name, 1, {
+          transport: 'proxy',
+        });
+      }
+    }
+  }
+
+  private addProxyRoute(route: ProxyRoute) {
+    this._expressRouter!.use(route.input.path, (req, res, next) => {
```

Comment:
route action is completely ignored, you register the route for every action here